src/main/groovy/com/github/spotbugs/snom/SpotBugsPlugin.java

```diff
@@ -39,8 +39,8 @@ public void apply(Project project) {
             javaBase -> {
               log.debug(
                   "The javaBase plugin has been applied, so making the check task depending on all of SpotBugsTask");
-              Task check = project.getTasks().getByName("check");
-              project.getTasks().withType(SpotBugsTask.class, check::dependsOn);
+              project.getTasks().named(JavaBasePlugin.CHECK_TASK_NAME).configure(task ->
```

Comment:
At this moment, we're sure that the `java-base` plugin has been applied to the project.  
Do we still need to defer the task creation?

Reply:
With `project.getTasks().getByName()` the `check` task is still created, see https://docs.gradle.org/current/userguide/task_configuration_avoidance.html for more info.

Reply:
ah I got the point. even though the plugin has been applied, it's still possible that user don't need it so we may skip creating the check task for better performance. thanks!